app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -554,18 +551,39 @@ public function hitEmail($stat, $request, $viaBrowser = false, $activeRequest =
         $this->em->persist($stat);
 
         // Flush the email stat entity in different transactions than the device stat entity to avoid deadlocks.
-        $this->flushAndCatch();
+        if ($throwDoctrineExceptions) {
+            $this->em->flush();
+        } else {
+            $this->flushAndCatch();
+        }
 
         if ($lead) {
-            $trackedDevice = $this->deviceTracker->createDeviceFromUserAgent($lead, $request->server->get('HTTP_USER_AGENT'));
+            $trackedDevice = $this->deviceTracker->createDeviceFromUserAgent(
+                $lead,
+                $request->server->get('HTTP_USER_AGENT')
+            );
+
+            // As the entity might be cached, present in EM, but not attached, we need to reload it
+            if (UnitOfWork::STATE_DETACHED === $this->em->getUnitOfWork()->getEntityState($trackedDevice)) {
```

Comment:
```suggestion
```
We can omit the condition as calling `$this->em->getRepository(LeadDevice::class)->find($trackedDevice->getId())` multiple times does not cause any issue.

Reply:
@fedys well this is the thing. there is some kind of caching here:
https://github.com/mautic/mautic/pull/12314/files#diff-09b3ad8df265e6fc5d43a0ead1c0d5394e085e56093f39af644b641d52827d82R566
if you refactor it not to cache or to reload it the entity on its own, it won't be needed. I will not refactor the method. This PR would never end. 
this condition makes it work less. no need to remove.

Reply:
yes, it is, the call is not necessary every time. Extra calls and extra populating the entity again for nothing. give me one reason to remove the condition.

Reply:
The thing is that if you call the `find()` method and the given entity **is already present in the unit of work**, the entity **is not populated again**. Instead the existing entity is returned right away. In other words the condition `if (UnitOfWork::STATE_DETACHED === $this->em->getUnitOfWork()->getEntityState($trackedDevice))` is done for you by Doctrine itself and thus unnecessary here.

Reply:
Let's keep the condition.

Reply:
I see, but the entity is in the UnitOfWork. it is just not attached and I am not sure if will get reloaded if it is there dettached. this is inspired but Doctrine's `refresh` method that has been deprecated as it can be done this way. if you are sure I WILL remove it.

Reply:
I'm 100% sure. Doctrine will not populate the entity again once it is in UoW and not detached. If it is detached and is in UoW it will be repopulated from the DB. We can safely remove the condition.

Reply:
@fedys forgot to thank you for teaching me this.